test/process_test.cc

```diff
@@ -204,6 +204,22 @@ TEST_P(ProcessTest, NoFlushWhenCancelExecutionBeforeLoadTestBegin) {
   EXPECT_EQ(numFlushes, 0);
 }
 
+TEST_P(ProcessTest, FailsIfUserDefinedOutputPluginSpecified) {
+  const std::string user_defined_output_plugin =
+      "{name:\"nighthawk.fake_user_defined_output\",typed_config:"
+      "{\"@type\":\"type.googleapis.com/nighthawk.FakeUserDefinedOutputConfig\"}}";
+  OptionsPtr options =
```

Comment:
Would std::unique_ptr<OptionsPtr> options(TestUtility::createOptionsImpl...()) more readable?

Reply:
Oops, I was about to say `std::unique_ptr<Options> options(TestUtility::createOptionsImpl...())`.

I didn't know the Envoy style is to use OptionsPtr. Your current code LGTM then.